gobblin-modules/gobblin-gitter/src/main/java/org/apache/gobblin/source/GitterMessageExtractor.java

```
@@ -0,0 +1,270 @@
+          for (JsonElement messageResponse : messagesResponse) {
+            log.info(String.format("Got message in json: %s", messageResponse));
+            JsonRecordAvroSchemaToAvroConverter<String> converter = new JsonRecordAvroSchemaToAvroConverter<>();
+            converter.init(workUnitState);
+            try {
+              GenericRecord messageAvro = converter.convertRecord(this.gitterMessageSchema,
+                  messageResponse.getAsJsonObject(), this.workUnitState).iterator().next();
+              if (null != messageAvro) {
+                log.info(String.format("Got message in avro: %s", messageResponse));
+                messageCollection.add(messageAvro);
+              } else {
+                log.error(String.format("Could not convert message: %s to Avro. Got null. Bailing out.", messageResponse));
+                throw new RuntimeException(String.format("Could not convert message: %s to Avro. Got null. Bailing out.",
+                    messageResponse));
+              }
+            } catch (DataConversionException dce) {
+              log.error(String.format("Could not convert message: %s to Avro. Bailing out.", messageResponse), dce);
+              throw new RuntimeException(dce);
+            }
+          }
+          break;
+        }
+        default: {
+          log.error(String.format("Could not fetch messages for room: %s "
+              + "Got response code: %s", this.gitterRoom, responseCode));
+          throw new RuntimeException(String.format("Could not fetch messages for room: %s "
+              + "Got response code: %s", this.gitterRoom, responseCode));
+        }
+      }
+    } catch (IOException | ParseException ex) {
+      log.error(String.format("Could not fetch messages for room: %s ", this.gitterRoom), ex);
+      throw new RuntimeException(ex);
+    }
+  }
+
+  @Override
+  public long getExpectedRecordCount() {
+    // We don't know this in advance
+    return 0;
```

If this is never used in Gitter, we can throw UnsupportedOperation exception.